### PR TITLE
Add xo-style commands to intkey CLI

### PR DESCRIPTION
**Pull request by nick-drozd**
This PR updates the intkey CLI to look like the xo CLI. The new commands are:

- `intkey set <name> <value>`
- `intkey inc <name> <value>`
- `intkey dec <name> <value>`
- `intkey show <name>`
- `intkey list`

~~All the old intkey commands have been removed except for workload, which will be removed when testing is changed over to smallbank.~~ The old intkey commands have been left alone. They can be removed later, but that will have to be done in conjunction with documentation updates.

### CODE REVIEW

File: sdk/examples/intkey_python/sawtooth_intkey/client_cli/intkey_cli.py [hunk 1]
```diff
@@ -0,0 +1,354 @@
...
+
+    parser.add_argument(
+        '--url',
+        type=str,
+        help='the url of the REST API')
+
+    parser.add_argument(
+        '--keyfile',
+        type=str,
+        help='the file in which the user\'s private key is stored')
```
Comment (chenette) [inline sdk/examples/intkey_python/sawtooth_intkey/client_cli/intkey_cli.py]: I suggest this rewording to shorten the content and avoid "in which":
     `the file containing the user\'s private key`

File: sdk/examples/intkey_python/sawtooth_intkey/client_cli/intkey_cli.py [hunk 2]
```diff
@@ -0,0 +1,354 @@
...
+
+    parser.add_argument(
+        '--wait',
+        nargs='?',
+        const=sys.maxsize,
+        type=int,
+        help='time to wait for transaction to commit')
```
Comment (chenette) [inline sdk/examples/intkey_python/sawtooth_intkey/client_cli/intkey_cli.py]: Need to specify the unit of time. I'm guessing that it's seconds. I prefer this version (similar to the rest_api wording):
     `Seconds to wait for the transaction to commit`

Also, there's some inconsistency across the CLIs - will this be standardized?
- seth has a short option as well as the long one (-w, --wait) 
- rest_api has a different name (-t TIMEOUT, --timeout TIMEOUT)

File: sdk/examples/intkey_python/sawtooth_intkey/client_cli/intkey_cli.py [hunk 3]
```diff
@@ -0,0 +1,354 @@
...
+
+    parser.add_argument(
+        '--url',
+        type=str,
+        help='the url of the REST API')
+
+    parser.add_argument(
+        '--keyfile',
+        type=str,
+        help='the file in which the user\'s private key is stored')
```
Comment (chenette) [inline sdk/examples/intkey_python/sawtooth_intkey/client_cli/intkey_cli.py]: (as above)

File: sdk/examples/intkey_python/sawtooth_intkey/client_cli/intkey_cli.py [hunk 4]
```diff
@@ -0,0 +1,354 @@
...
+
+    parser.add_argument(
+        '--wait',
+        nargs='?',
+        const=sys.maxsize,
+        type=int,
+        help='time to wait for transaction to commit')
```
Comment (chenette) [inline sdk/examples/intkey_python/sawtooth_intkey/client_cli/intkey_cli.py]: (as above)

File: sdk/examples/intkey_python/sawtooth_intkey/client_cli/intkey_cli.py [hunk 5]
```diff
@@ -0,0 +1,354 @@
...
+
+    parser.add_argument(
+        '--url',
+        type=str,
+        help='the url of the REST API')
+
+    parser.add_argument(
+        '--keyfile',
+        type=str,
+        help='the file in which the user\'s private key is stored')
```
Comment (chenette) [inline sdk/examples/intkey_python/sawtooth_intkey/client_cli/intkey_cli.py]: (as above)

File: sdk/examples/intkey_python/sawtooth_intkey/client_cli/intkey_cli.py [hunk 6]
```diff
@@ -0,0 +1,354 @@
...
+
+    parser.add_argument(
+        '--wait',
+        nargs='?',
+        const=sys.maxsize,
+        type=int,
+        help='time to wait for transaction to commit')
```
Comment (chenette) [inline sdk/examples/intkey_python/sawtooth_intkey/client_cli/intkey_cli.py]: (as above)

File: sdk/examples/intkey_python/sawtooth_intkey/client_cli/intkey_cli.py [hunk 7]
```diff
@@ -0,0 +1,354 @@
...
+    print(response)
+
+
+def add_show_parser(subparsers, parent_parser):
+    parser = subparsers.add_parser('show', parents=[parent_parser])
+
+    parser.add_argument(
+        'name',
+        type=str,
+        help='the name of the value to show')
```
Comment (chenette) [inline sdk/examples/intkey_python/sawtooth_intkey/client_cli/intkey_cli.py]: 👍  Thanks for not saying "print".

File: sdk/examples/intkey_python/sawtooth_intkey/client_cli/intkey_cli.py [hunk 8]
```diff
@@ -0,0 +1,354 @@
...
+
+    parser.add_argument(
+        'name',
+        type=str,
+        help='the name of the value to decrement')
+
+    parser.add_argument(
+        'value',
+        type=int,
+        help='the value by which to decrement')
```
Comment (chenette) [inline sdk/examples/intkey_python/sawtooth_intkey/client_cli/intkey_cli.py]: I'd prefer to avoid "by which".  How about:
     `the amount to decrement`

File: sdk/examples/intkey_python/sawtooth_intkey/client_cli/intkey_cli.py [hunk 9]
```diff
@@ -0,0 +1,354 @@
...
+    print(response)
+
+
+def add_dec_parser(subparsers, parent_parser):
+    parser = subparsers.add_parser('dec', parents=[parent_parser])
+
+    parser.add_argument(
+        'name',
+        type=str,
+        help='the name of the value to decrement')
```
Comment (chenette) [inline sdk/examples/intkey_python/sawtooth_intkey/client_cli/intkey_cli.py]: (as above)

File: sdk/examples/intkey_python/sawtooth_intkey/client_cli/intkey_cli.py [hunk 10]
```diff
@@ -0,0 +1,354 @@
...
+    print(response)
+
+
+def add_inc_parser(subparsers, parent_parser):
+    parser = subparsers.add_parser('inc', parents=[parent_parser])
+
+    parser.add_argument(
+        'name',
+        type=str,
+        help='the name of the value to increment')
```
Comment (chenette) [inline sdk/examples/intkey_python/sawtooth_intkey/client_cli/intkey_cli.py]: The word "value" is confusing here. I assume this is the item (variable?) to be modified, not the amount to add?  If so, how about this wording:
     `the variable name of the entry to be modified`

I'm not sure if "entry" is the correct term, but it's used in section 9.4, IntegerKey Transaction Family Specification.

File: sdk/examples/intkey_python/sawtooth_intkey/client_cli/intkey_cli.py [hunk 11]
```diff
@@ -0,0 +1,354 @@
...
+
+    parser.add_argument(
+        'value',
+        type=int,
+        help='the value by which to increment')
```
Comment (chenette) [inline sdk/examples/intkey_python/sawtooth_intkey/client_cli/intkey_cli.py]: I'd prefer to avoid "by which".  How about:
     `the amount to increment`

File: sdk/examples/intkey_python/sawtooth_intkey/client_cli/intkey_cli.py [hunk 12]
```diff
@@ -0,0 +1,354 @@
...
+    return parser
+
+
+def add_set_parser(subparsers, parent_parser):
+    parser = subparsers.add_parser('set', parents=[parent_parser])
+
+    parser.add_argument(
+        'name',
+        type=str,
+        help='the name of the value to set')
```
Comment (chenette) [inline sdk/examples/intkey_python/sawtooth_intkey/client_cli/intkey_cli.py]: The word "value" is confusing here, especially because "value" means something different for the next argument.  How about "item"? (Or use a more specific term if there is one.)

File: sdk/examples/intkey_python/sawtooth_intkey/client_cli/intkey_cli.py [hunk 13]
```diff
@@ -0,0 +1,353 @@
...
+
+    parser.add_argument(
+        'name',
+        type=str,
+        help='the name of the key to increment')
+
+    parser.add_argument(
+        'value',
+        type=int,
+        help='the value to increment')
```
Comment (chenette) [inline sdk/examples/intkey_python/sawtooth_intkey/client_cli/intkey_cli.py]: Minor comment: Because the argument name is "value", I'd prefer "the amount to increment" (or other word that isn't "value").  Tech-writer best practice is to avoid circular definitions by using a different word in the description. 

This comment also applies to "set value" above and "decrement value" below.

**Comment by chenette** [comment]
@peterschwarz:The plan, I believe, is to work on better (and consistent) CLI usage text as a separate task. I am currently gathering recommendations and will create a CLI usage style guide as part of https://jira.hyperledger.org/browse/STL-744.

**Comment by nick-drozd** [comment]
@chenette @peterschwarz I added some not-very-thoughtful usage text descriptions to the new commands. We can discuss better alternatives now or come back to it later.
